### ISSUE TITLE
UIGraphicsGetCurrentContext() returns nil - what now?

### ISSUE THREAD

**Issue by jboegeholz** [fulldecent/FDWaveformView#84]
Hi!

I've successfully integrated your library into one of my projects. It works very nice, but sometimes the waveform is not rendered and I get the "FDWaveformView failed to get graphics context".

Do You know what to do then or the circumstances why it can happen?

BR 
Joern

**Comment by fulldecent** [comment]
Hello Joern, thank you for sharing. 

Hmm... no, I have never seen this error before. Can you please share what hardware and iOS versions and is the simulator or device? Also is there jailbreak or anything else weird going on here?

**Comment by jboegeholz** [comment]
Hi William! I've tested on iPhone6 with iOS 10.2.1 as device

**Comment by fulldecent** [comment]
Sorry, I cannot reproduce this. I am using the example app on simulator, iPhone and iPad successfully.

Are you able to successfully use the example app?

If so, are you able to slowly introduce changes in the working example to make it break?

**Comment by fulldecent** [comment]
Has anyone tried to bisect this regression to find out where it was introduced? We had a big change in #79, maybe it was then.

**Comment by jboegeholz** [comment]
I've tried to reproduce this issue with 2.1.0 and it seems to be fixed :-)

**Comment by fulldecent** [comment]
Excellent, thanks for testing.

**Comment by DBitros** [comment]
I seem to be having the same issue using 3.0, some waveforms rendering and other do not

**Comment by fulldecent** [comment]
Hmm, is this reproducible for certain input files?

Also, please confirm you are using a real device. Simulator is known to be weird. (Maybe we can add that as a console log note!)

**Comment by DBitros** [comment]
I've just tested this on an iPhone 6 device and I am seeing the same issue.

I'm displaying .m4a files. 

To provide some more information that might help: 

The FDWaveformView is placed in a custom UTableViewCell, in a table that loads the .m4a files from a local file which has been downloaded from a server. Some of these files load a waveform and others do not. All are .m4a and all are downloaded from the same call from the server. 

I've done a quick test and loaded a .m4a file I downloaded from the server and directly added it to the bundle within the project and tried loading it this way and it does not show the waveform either on load. I have then converted this file same file to .mp3 and loaded from the bundle and it works. The file I've downloaded does not seem to be corrupted as I can play it via iTunes.
